bip-0099.mediawiki

```diff
@@ -0,0 +1,333 @@
+<pre>
+  BIP: 99
+  Title: Motivation and deployment of consensus rule changes ([soft/hard]forks)
+  Author: Jorge Timón [jtimon@jtimon.cc]
+  Status: Draft
+  Type: Informational | Process
+  Created: 2015-06-20
+  Post-History: http://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-June/008936.html
+</pre>
+
+==Abstract==
+
+This BIP attempts to create a taxonomy of the different types of
+consensus forks and proposes a deployment mechanism for each of them.
+
+==Motivation==
+
+The security assumptions of p2p consensus-based systems like Bitcoin are
+not always well understood, and the upgrade mechanisms to the
+consensus rules are a work in progress that may vary depending on the
+type of change being deployed.
+Discussing such changes without a uniform view on the deployment
+paths often leads to misunderstandings and unnecessarily delays the
+deployment of changes. 
+
+==Defintions==
+
+- Software fork: in free software, an existing project can be copied
+  and extended/maintained without permission from the original
+  project's maintainers.
+- Consensus fork: a divergence in the implementation of the
+  verification rules can impede the expected eventual convergence of
+  the network in a single chain that has the most proof of work and
+  also satisfies the rules. This can be intentional or be caused by a
+  bug in consensus validation reimplementations.
+- Softfork: an intentional consensus fork where everything that
+  was previously invalid remains invalid, but some things that were
+  valid also become invalid. Majority of miners can impose
+  this kind of changes and they also have some deployment advantages.
+- Hardfork: an intentional consensus fork that is not a softfork.
```

Comment:
There can be non-intentional hard-forks, as evidenced by the existence of previous chain consensus forking. Non-intentional hard-forks are also not soft-forks, in addition to intentional consensus forks.